### PR TITLE
Organize documentation index

### PR DESCRIPTION
**Pull request by mrocklin**
This splits the index into three sections:

1.  Get Started
2.  Use
3.  Develop

It also removes the index from the main body of the page, and instead
relies on the sidebar

### CODE REVIEW

**Comment by mrocklin** [comment]
OK to merge @TomAugspurger ?

File: docs/source/auto_examples/index.rst [hunk 1]
```diff
@@ -92,13 +92,13 @@ This is a set of runnable Jupyter notebooks.
...
 
-    :download:`Download all examples in Python source code: auto_examples_python.zip <//Users/taugspurger/sandbox/dask-ml/docs/source/auto_examples/auto_examples_python.zip>`
+    :download:`Download all examples in Python source code: auto_examples_python.zip <//home/mrocklin/workspace/dask-ml/docs/source/auto_examples/auto_examples_python.zip>`
 
 
 
   .. container:: sphx-glr-download
 
-    :download:`Download all examples in Jupyter notebooks: auto_examples_jupyter.zip <//Users/taugspurger/sandbox/dask-ml/docs/source/auto_examples/auto_examples_jupyter.zip>`
+    :download:`Download all examples in Jupyter notebooks: auto_examples_jupyter.zip <//home/mrocklin/workspace/dask-ml/docs/source/auto_examples/auto_examples_jupyter.zip>`
```
Comment (mrocklin) [inline docs/source/auto_examples/index.rst]: Hrm, this is an odd change.  This is an auto-generated file?  Presumably rtd will fill this in?

Comment (TomAugspurger) [reply docs/source/auto_examples/index.rst]: Yeah... I'm not sure how but it seems to work.

I haven't found a way of doing these examples that I like. Some of the examples require too much memory / take too long for RTD to build them, so we have to generate them ahead of time.

**Comment by mrocklin** [comment]
FYI I just pushed 6a16212 to master.  My apologies for the lack of process.

**Comment by TomAugspurger** [comment]
👍

**Pull request merged**